core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -987,7 +987,7 @@ class ReplicaManager(val config: KafkaConfig,
 
         // If we are the leader, determine the preferred read-replica
         val preferredReadReplica = clientMetadata.flatMap(
-          metadata => findPreferredReadReplica(tp, metadata, replicaId, fetchInfo.fetchOffset, fetchTimeMs))
+          metadata => findPreferredReadReplica(partition, metadata, replicaId, fetchInfo.fetchOffset, fetchTimeMs))
```

Comment:
Partition is already accessible, there's no need to look it it up in findPreferredReadReplica.